pagopa/io-app
# chore: [IOBP-74] Transaction list item redesign

## Short description

added new `TransactionListItem` generic component
<img width="250" alt="Screenshot 2023-06-27 at 12 20 17" src="https://github.com/pagopa/io-app/assets/60693085/6424bdf5-ed34-4fc9-a4da-60a2f8aeb30b">

## List of changes proposed in this pull request
- added `TransactionListItem` component, along with its loading state
- added generic `PressableListItemBase` component and ` useDefaultListItemSpringAnimation` hook
- added required I18n keys
- added the component and its variants in the design system's `list items` section

## How to test
navigate to the DS playground, in the listItems section and make sure everything renders correctly and follows the design
